cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataEntity.java

```diff
@@ -49,9 +49,10 @@
  *    .appendAsType("field", "myField").build();
  * </pre>
  * <p>
- * Every metadata entity has a type. The type can be changed during append by using
- * {@link MetadataEntity.Builder#appendAsType(String, String)} which does not necessarily have to be the last append.
- * In some cases, the type is a key in middle. An example of this is ApplicationId which ends with "version"
+ * Every metadata entity has a type. If a type is not specifically specified by calling
```

Comment:
Can you add a sentence to the first paragraph that keys are non case-sensitive?

Reply:
Done.